bot/extensions/reminder_cog.py

```diff
@@ -0,0 +1,128 @@
+            self.bot.scheduler.remove_job(job.id)
+
+    def _build_embed(self, title: str, message: str, author: str) -> Embed:
+        """Builds a Discord embed with the given description.
+
+        :param title: The title of the embed.
+        :param message: The description/message of the embed.
+        :param author: The author of the embed.
+
+        :return: The constructed Embed object.
+        """
+        embed = Embed(
+            color=self.bot.default_color,
+            title=f"**{title}**",
+            description=message,
+            timestamp=datetime.now(),
+        )
+        embed.set_author(name=author)
+        embed.set_thumbnail(
+            url=(
+                "https://external-content.duckduckgo.com/iu/?u="
+                "https%3A%2F%2Fstatic.vecteezy.com%2Fsystem%2Fresources%2Fpreviews"
+                "%2F012%2F067%2F332%2Foriginal%2Fhand-holding-a-stopwatch-timer-png.png"
+                "&f=1&nofb=1"
+            )
+        )
+        return embed
+
+    def _build_timer(self, match: Match[str]) -> timedelta:
+        """Converts the parsed time format into a timedelta.
+
+        :param match: The timer from the user containing amount and unit.
+                    amount: The amount of time before the reminder.
+                    unit: The unit of time (s=second, m=minute, h=hour, d=day).
+
+        :return: The constructed timedelta for the reminder.
+        """
+        amount, unit = match.groups()
+        amount = int(amount)
+        match unit:
+            case "s":
+                return timedelta(seconds=amount)
+            case "m":
+                return timedelta(minutes=amount)
+            case "h":
+                return timedelta(hours=amount)
+            case "d":
+                return timedelta(days=amount)
+
+    @hybrid_command(
+        name="reminder", help="Set a reminder with a message", usage="{timer} {message}"
+    )
+    async def reminder(self, ctx, timer: str, *, message: str) -> None:
+        """
+        Set a reminder for the user.
+
+        :param ctx: Command context.
+        :param timer: Time after which to remind the user such as '10m', '2h', '1d'.
+        :param message: The reminder message.
+        """
+        time_pattern = re.compile(r"(\d+)([smhd])")
+        match = time_pattern.fullmatch(timer)
+        if not match:
+            await ctx.send(
+                "Invalid time format! Use '10m' for 10 minutes,'2h' for 2 hours, etc."
+            )
+            return
+
+        reminder_delta = self._build_timer(match)
+        reminder_time = datetime.now(self.timezone) + reminder_delta
+
+        self.jobs.append(
+            self.bot.scheduler.add_job(
+                self.send_reminder,
+                "date",
+                run_date=reminder_time,
+                args=[ctx, message],
+                id=f"reminder_{ctx.author.id}_{datetime.now().timestamp()}",
+            )
+        )
+
+        reminder_time = (
```

Comment:
There's an issue with user timezone. It doesn't work super well on staging. It would be better to use discord relative date format like the time_cog does.